src/pages/release-codes/forms/ReleaseCodeForm.js

```diff
@@ -1,86 +1,70 @@
 // ** MUI Imports
 import { Grid } from '@mui/material'
-import { useContext, useEffect, useState } from 'react'
-import { useFormik } from 'formik'
+import { useContext, useEffect } from 'react'
+import { useForm } from 'src/hooks/form'
 import * as yup from 'yup'
 import FormShell from 'src/components/Shared/FormShell'
 import toast from 'react-hot-toast'
 import { RequestsContext } from 'src/providers/RequestsContext'
 import { useInvalidate } from 'src/hooks/resource'
 import { ResourceIds } from 'src/resources/ResourceIds'
-
-// ** Custom Imports
 import CustomTextField from 'src/components/Inputs/CustomTextField'
-import CustomTextArea from 'src/components/Inputs/CustomTextArea'
-
 import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
 import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
 import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
 
 export default function ReleaseCodeForm({ labels, maxAccess, recordId }) {
-  const [isLoading, setIsLoading] = useState(false)
-  const [editMode, setEditMode] = useState(!!recordId)
-
-  const [initialValues, setInitialData] = useState({
-    recordId: null,
-    name: '',
-    reference: ''
-  })
-
+  const { platformLabels } = useContext(ControlContext)
   const { getRequest, postRequest } = useContext(RequestsContext)
 
-  //const editMode = !!recordId
+  const editMode = !!recordId
```

Comment:
it's better to replace by `const editMode = !formik.values.editMode ` since after saving directly in new mode recordId not updated

Reply:
const editMode = !!formik.values.recordId fixed